test/test.go

```diff
@@ -123,22 +123,22 @@ func GenerateIDs(n int) []*key.Pair {
 }
 
 // BatchIdentities generates n insecure identities
-func BatchIdentities(n int) ([]*key.Pair, *key.Group) {
+func BatchIdentities(n int, decouplePrevSig bool) ([]*key.Pair, *key.Group) {
```

Comment:
This should not need to change - you can set the field afterwards if you want, only for V2, but the rest of the test could stay the same. I mean it's not wrong but again it feels there is a lot of small changes because the API changed while we could simply create a new API for v2, as in a new protocol and let the old one pretty similar.
Same comment for beacon.Message()